File: verifier-problems/src/main/java/com/jetbrains/pluginverifier/persistence/IdeVersionTypeAdapter.java
package com.jetbrains.pluginverifier.persistence;

import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import com.intellij.structure.domain.IdeVersion;

import java.io.IOException;

/**
 * @author Sergey Patrikeev
 */
class IdeVersionTypeAdapter extends TypeAdapter<IdeVersion> {

  @Override
  public void write(JsonWriter out, IdeVersion value) throws IOException {
    out.value(value.asString());
  }

  @Override
  public IdeVersion read(JsonReader in) throws IOException {
    return IdeVersion.createIdeVersion(in.nextString());
  }

}

File: verifier-problems/src/main/kotlin/com/jetbrains/pluginverifier/persistence/IdeVersionTypeAdapter.kt
package com.jetbrains.pluginverifier.persistence

import com.google.gson.TypeAdapter
import com.google.gson.stream.JsonReader
import com.google.gson.stream.JsonWriter
import com.intellij.structure.domain.IdeVersion

import java.io.IOException

/**
 * @author Sergey Patrikeev
 */
class IdeVersionTypeAdapter : TypeAdapter<IdeVersion>() {

  @Throws(IOException::class)
  override fun write(out: JsonWriter, value: IdeVersion) {
    out.value(value.asString())
  }

  @Throws(IOException::class)
  override fun read(`in`: JsonReader): IdeVersion {
    return IdeVersion.createIdeVersion(`in`.nextString())
  }

}
